### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+| `global_cache_dir`                | :heavy_check_mark:       |              | Cache directory to download Mattermost files to                                                                                                                                                        |
+| `roundcube_sql_password`              | :heavy_check_mark: |       | Password of the `roundcube_sql_user`                                                                                                                                                                          |
+| `roundcube_des_key`              | :heavy_check_mark: |  | This key is used to encrypt the users imap password which is stored in the session record (and the client cookie if remember password is enabled). Please provide a string of exactly 24 chars.                                                                                                                                                                                |
```
Comment (SlothOfAnarchy) [inline README.md]: please also ensure that the key has this length

File: README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+| `roundcube_sql_user`             | :heavy_multiplication_x: |        `roundcube`      | Database user                                                                                                                                                                     |
+| `roundcube_sql_database`               | :heavy_multiplication_x: |              | Database name                                                                                                                                                               |
+| `roundcube_default_host`  | :heavy_check_mark:       |     `ssl://{{ mailserver_fqdn_hostname | default('localhost') }}`         | Imap server                                                                                                                                                            |
+| `roundcube_default_port`     | :heavy_check_mark:       |       `993`       | Imap port                                                                                                                                                            |
```
Comment (SlothOfAnarchy) [inline README.md]: maybe rename this
from the name I would think this is the port where I can reach roundcube
same for host

File: README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+| `roundcube_sql_host`                | :heavy_multiplication_x: |        `localhost`      | Host of for the database                                                                                                                                                                |
+| `roundcube_sql_user`             | :heavy_multiplication_x: |        `roundcube`      | Database user                                                                                                                                                                     |
+| `roundcube_sql_database`               | :heavy_multiplication_x: |              | Database name                                                                                                                                                               |
+| `roundcube_default_host`  | :heavy_check_mark:       |     `ssl://{{ mailserver_fqdn_hostname | default('localhost') }}`         | Imap server                                                                                                                                                            |
```
Comment (SlothOfAnarchy) [inline README.md]: why is this required when there is a default?
Same for the other vars below.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+| `roundcube_default_port`     | :heavy_check_mark:       |       `993`       | Imap port                                                                                                                                                            |
+| `roundcube_smtp_server`     | :heavy_check_mark:       |       `993`       | smtp server                                                                                                                                                          |
```
Comment (SlothOfAnarchy) [inline README.md]: I think you switched the default for smtp port and server

File: README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+| `roundcube_smtp_user`     | :heavy_check_mark:       |       `%u`       | smtp user                                                                                                                                                          |
+| `roundcube_smtp_pass`     | :heavy_check_mark:       |       `%p`       | smtp password                                                                                                                                                            |
+| `roundcube_support_url`     | :heavy_check_mark:       |       ``       | Provide an URL where a user can get support for this Roundcube installation.                                                                                                                                                            |
+| `roundcube_ip_check`     | :heavy_check_mark:       |       `true`       | smtp password                                                                                                                                                            |
```
Comment (SlothOfAnarchy) [inline README.md]: description?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+| `roundcube_mail_pagesize`     | :heavy_check_mark:       |       `50`       | Mails to be displayed on one page                                                                                                                                                            |
+| `roundcube_draft_autosave`     | :heavy_check_mark:       |       `300`       | After how many seconds roundcube should do an autosave                                                                                                                                                            |
+| `roundcube_mime_param_folding`     | :heavy_check_mark:       |       `0`       |                                                                                Encoding of long/non-ascii attachment names                                                                            |
+| `roundcube_mdn_requests`     | :heavy_check_mark:       |       `2`       | Behavior if a received message requests a message delivery notification (read receipt)                                                                                                                                                            |
```
Comment (SlothOfAnarchy) [inline README.md]: What is the behaviour for `2`, what are the options?

File: tasks/configure.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+
```
Comment (SlothOfAnarchy) [inline tasks/configure.yml]: delete line

File: tasks/configure.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+- name: Add roundcube config
```
Comment (SlothOfAnarchy) [inline tasks/configure.yml]: Copy* ?

File: tasks/database.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  register: _roundcube_db
+  changed_when: false
+
+- name: Import database
+  shell: 'mysql {{ roundcube_sql_database }} < {{ roundcube_unarchive_dir }}/SQL/mysql.initial.sql'
+  when: _roundcube_db.stdout == '0'
+
+- name: Run update
+  shell: '{{ roundcube_unarchive_dir }}/bin/update.sh'
+  changed_when: false # make it idempotence
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]: *idempotent
But why set this at all? An update will change things, can't you parse the script output?

File: tasks/database.yml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  tags: ['skip_ansible_lint']
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]: why this tag? I would remove it.

File: tasks/install.yml [hunk 11]
```diff
@@ -0,0 +1,101 @@
+---
+- name: Install required apt packages
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Add update cache task that is not changed

File: tasks/install.yml [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+- name: Ensure working dir exists
+  file:
+    path: '{{ roundcube_working_dir }}'
+    state: directory
+
+- name: 'Download roundcube from {{ roundcube_download_url }}'
+  get_url: 
+    url: '{{ roundcube_download_url }}'
+    dest: "{{ global_cache_dir | mandatory }}/roundcubemail-{{ roundcube_install_version }}.tar.gz"
+  when: mattermost_enterprise
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: no

File: tasks/install.yml [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+    owner: '{{ roundcube_user }}'
+    group: '{{ roundcube_group }}'
+    mode: '0711'
+  with_items:
+    - '{{ roundcube_working_dir }}'
+    - '{{ roundcube_working_dir }}/logs'
+    - '{{ roundcube_working_dir }}/sessions'
+    - '{{ roundcube_working_dir }}/usr/share/'
+
+- name: Prepare directories
```
Comment (SlothOfAnarchy) [inline tasks/install.yml:70]: duplicate

File: tasks/install.yml [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    - '{{ roundcube_working_dir }}/tmp'
+    - '{{ roundcube_unarchive_dir }}/logs'
+    - '{{ roundcube_unarchive_dir }}/temp'
+
+- name: Check if composer.json-dist exists
+  stat:
+    path: '{{ roundcube_unarchive_dir }}/composer.json-dist'
+  register: _roundcube_composer_json
+
+- name: Copy missing composer.json
```
Comment (SlothOfAnarchy) [inline tasks/install.yml:87]: Why not always copy the file?
also, ansible can copy files, you don't need `cp`

File: tasks/install.yml [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+    creates: '{{ roundcube_unarchive_dir }}/composer.json'
+  when: _roundcube_composer_json.stat.exists
+  become: true
+  become_user: '{{ roundcube_user }}'
+
+- name: Run 3rd party installer
+  shell: "./bin/install-jsdeps.sh"
+  args: 
+    chdir: "{{ roundcube_unarchive_dir }}"
+  changed_when: false
```
Comment (SlothOfAnarchy) [inline tasks/install.yml:101]: Can't you parse the output or return code for changes?
Shouldn't this maybe only be run after an update?

File: tasks/install.yml [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+- name: Ensure git is installed
+  apt:
+    pkg: git
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Move git to the install list above

File: tasks/install.yml [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+    state: present
+
+- name: Run composer install
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Use more meaningful title e.g. "Install php plugins"

File: tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,10 @@
+---
+
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: remove empty line

File: templates/config.inc.php [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+$config['support_url'] = '{{ roundcube_support_url }}';
+
+// Name your service. This is displayed on the login screen and in the window title
+$config['product_name'] = '{{ roundcube_product_name }}';
+
+// this key is used to encrypt the users imap password which is stored
+// in the session record (and the client cookie if remember password is enabled).
+// please provide a string of exactly 24 chars.
+// YOUR KEY MUST BE DIFFERENT THAN THE SAMPLE VALUE FOR SECURITY REASONS
+$config['des_key'] = '{{ roundcube_des_key | mandatory }}';
```
Comment (SlothOfAnarchy) [inline templates/config.inc.php]: You don't need `mandatory` in jinja, only in tasks

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/configure.yml (hunk 7 above)
Comment (Scriptkiddi) [reply tasks/configure.yml]: fixed

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: tasks/configure.yml (hunk 8 above)
Comment (Scriptkiddi) [reply tasks/configure.yml]: fixed

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: tasks/database.yml (hunk 10 above)
Comment (Scriptkiddi) [reply tasks/database.yml]: fixed

File: tasks/install.yml (hunk 11 above)
Comment (Scriptkiddi) [reply tasks/install.yml]: fixed

File: tasks/install.yml (hunk 13 above)
Comment (Scriptkiddi) [reply tasks/install.yml:70]: no

Comment (Scriptkiddi) [reply tasks/install.yml:70]: different permissions

File: tasks/install.yml (hunk 15 above)
Comment (Scriptkiddi) [reply tasks/install.yml:101]: does not matter the script determins if it needs to update anything

File: tasks/install.yml (hunk 16 above)
Comment (Scriptkiddi) [reply tasks/install.yml]: done

File: tasks/install.yml (hunk 17 above)
Comment (Scriptkiddi) [reply tasks/install.yml]: fixed

File: tasks/main.yml (hunk 18 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: templates/config.inc.php (hunk 19 above)
Comment (Scriptkiddi) [reply templates/config.inc.php]: fixed

File: tasks/install.yml [hunk 20]
```diff
@@ -97,7 +105,7 @@
     pkg: git
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: The task is still there though

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

**Review by SlothOfAnarchy — APPROVED** [review]
